packages/astro/src/core/render/route-cache.ts

```diff
@@ -102,14 +102,14 @@ export class RouteCache {
 		// NOTE: This shouldn't be called on an already-cached component.
 		// Warn here so that an unexpected double-call of getStaticPaths()
 		// isn't invisible and developer can track down the issue.
-		if (this.mode === 'production' && this.cache[route.component]?.staticPaths) {
-			this.logger.warn(null, `Internal Warning: route cache overwritten. (${route.component})`);
+		if (this.mode === 'production' && this.cache[route.route]?.staticPaths) {
+			this.logger.warn(null, `Internal Warning: route cache overwritten. (${route.route})`);
```

Comment:
I think this will switch one conflict for another. Imagine a project with two integrations such that:
- Integration A injects route `/[...slug]` pre-rendered that generates the paths `/foo` and `/bar`
- Integration B injects route `/[...slug]` pre-rendered that generates the paths `/baz` and `/qux`
- A `src/pages/[...slug].astro` file in the project.

Those three are the same route, so they would conflict in the cache. Two integrations may define the same route for different components, and a single integration may define the same component for different routes.

Maybe the cache key should be the combination of both

Reply:
> Maybe the cache key should be the combination of both

I thought about it too! And thank you for the comment, it's good to see that I'm not the only one :) I'll apply the change